### PR TITLE
[new release] mirage, mirage-runtime, functoria and functoria-runtime (4.2.1)

### PR DESCRIPTION
**Pull request by hannesm**
The MirageOS library operating system

- Project page: <a href="https://github.com/mirage/mirage">https://github.com/mirage/mirage</a>
- Documentation: <a href="https://mirage.github.io/mirage/">https://mirage.github.io/mirage/</a>

#### Fixed

- In the generated opam file, also run "depext-lockfile" (mirage/mirage#1342, @hannesm)

### CODE REVIEW

File: packages/functoria/functoria.4.2.1/opam [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["env" "INSIDE_FUNCTORIA_TESTS=1" "dune" "exec" "-p" name "-j" jobs "--"
+     "test/functoria/e2e/test.exe"] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.9.0" | (with-test & >= "3.0.0")}
```
Comment (kit-ty-kate) [inline packages/functoria/functoria.4.2.1/opam:26]: I don't think `| with-test & other-constraint` is what you want. If you substitute you get:
* "dune" {>= "2.9.0" | false} when with-test is false
* "dune" {>= "2.9.0" | >= "3.0.0") when with-test is true, which simplifies into `"dune" {>= "2.9.0"}`
```suggestion
  "dune" {>= "2.9.0"}
  "dune" {with-test & >= "3.0.0"}
```

File: packages/mirage/mirage.4.2.1/opam [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test & os != "macos"}
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.9.0" | (with-test & >= "3.0.0")}
```
Comment (kit-ty-kate) [inline packages/mirage/mirage.4.2.1/opam:22]:
```suggestion
  "dune" {>= "2.9.0"}
  "dune" {with-test & >= "3.0.0"}
```

**Comment by hannesm** [comment]
Thanks, I pushed the suggested changes back https://github.com/mirage/mirage/commit/f79387d95cc8641376e8a299f6625d3a19bc3e18

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**